python/GangaLHCb/Lib/Applications/BenderBox.py

```diff
@@ -0,0 +1,528 @@
+    - other parameters  can be added as keyword arguments 
+    >>> j = Job ( application = prepareBender ( 'v30r1' , module = 'the_path/the_modue.py' , use_tmp = True , params = {} , events = 100 ) ) 
+    """
+    if use_tmp or not path :
+        path = tempfile.mkdtemp ( prefix = 'GANGA_' )
+        logger.info ( 'Bender application will be prepared in the temporary directory %s' % path )
+        
+    the_path = prepare_cmake_app ( 'Bender' , version , path )
+    
+    from Ganga.GPI import BenderModule as _BM  ## black magic
+    return _BM ( module = module , directory = the_path , **kwargs )
+
+# =============================================================================
+## prepare BenderRun application
+#  - specify the script 
+#  @code
+#  j = Job ( application = prepareBenderRun ( 'v30r1' , script = 'the_path/the_script.py') ) 
+#  @endcode
+#  - specify the scripts 
+#  @code
+#  j = Job ( application = prepareBenderRun ( 'v30r1' , scripts = ['the_path/the_script.py' , 'another_script.py' ] ) ) 
+#  @endcode
+#  - specify the directory explicitely:
+#  @code
+#  j = Job ( application = prepareBenderRun ( 'v30r1' , path = '$HOME/mydir' , script = 'the_script.py' ) 
+#  @endcode
+#  - use the temporary directory 
+#  @code
+#  j = Job ( application = prepareBenderRun ( 'v30r1' , use_tmp = True , script = 'the_script.py' )
+#  @endcode
+#  - one can also use the configuration parameters:
+#  @code
+#  j = Job ( application = prepareBender ( 'v30r1' , script = 'the_path/the_script.py' , commands = [] , imports = []  ) ) 
+#  @endcode
+def prepareBenderRun ( version , script = '' , path = '$HOME/cmtuser', use_tmp = False , **kwargs ) :
+    """Prepare BenderRun application
+    - specify the script 
+    >>> j = Job ( application = prepareBenderRun ( 'v30r1' , script = 'the_path/the_script.py') ) 
+    - specify the scripts 
+    >>> j = Job ( application = prepareBenderRun ( 'v30r1' , scripts = ['the_path/the_script.py' , 'another_script.py' ] ) ) 
+    - specify the directory explicitely:
+    >>> j = Job ( application = prepareBenderRun ( 'v30r1' , path = '$HOME/mydir' , script = 'the_script.py' ) 
+    - use the temporary directory 
+    >>> j = Job ( application = prepareBenderRun ( 'v30r1' , use_tmp = True , script = 'the_script.py' )
+    One can also use the configuration parameters:
+    >>> j = Job ( application = prepareBender ( 'v30r1' , script = 'the_path/the_script.py' , commands = [] , imports = []  ) ) 
+    """
+    if use_tmp or not path :
+        path = tempfile.mkdtemp ( prefix = 'GANGA_' )
+        logger.info ( 'Bender application will be prepared in the temporary directory %s' % path )
+        
+    the_path = prepare_cmake_app ( 'Bender' , version , path )
+    #
+    scripts = kwarsg.pop('scripts', [] )
+    if script : scripts.append ( script )
+    #
+    from Ganga.GPI import BenderRun as _BR  ## black magic    
```

Comment:
fyi this _magic_ it's because you're configuring a proxied object which has extra wrappers around the get/set/str and a few other methods over an un-wrapped `GangaObject`.
If you return a properly configured object you can also do an `return addProxy(_BR(...))` but this is slightly less intuitive to what these functions are doing (just reducing the amount of code the user has to type).